main.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+// Great help
+// https://golangcode.com/how-to-read-a-csv-file-into-a-struct/
+
+import (
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+)
+
+const csvFilename = "pokemons.csv"
+
+type Pokemon struct {
+	Id   int    `json:"id"`
+	Name string `json:"name"`
+}
+
+type DB struct {
+	data map[int]Pokemon
+}
+
+func main() {
+	pokemonDB := new(DB)
+	pokemonDB.initFromCsv(csvFilename)
+
+	// Loop through pokemons and print 'em all
+	for _, pokemon := range pokemonDB.data {
+		fmt.Printf("ID: %v, Name %v\n", pokemon.Id, pokemon.Name)
+	}
+
+	// Configure router
+	router := gin.Default()
+	router.GET("/pokemons", pokemonDB.getAllPokemons)
+	router.GET("/pokemons/:id", pokemonDB.getPokemonById)
+
+	// Start server
+	router.Run("localhost:8082")
+}
+
+// TODO: I do not like mixing DB and GIN context
```

Comment:
Applying Clean Architecture you will do it.
I think in the third workshop, we will explain a little more.
If you have any questions meanwhile, just ping me, we could have a 1:1 mentoring.